src/josepy/interfaces.py

```diff
@@ -10,6 +10,8 @@
 # pylint: disable=no-self-argument,no-method-argument,no-init,inherit-non-class
 # pylint: disable=too-few-public-methods
 
+JSONDeSerializableTypeVar = TypeVar("JSONDeSerializableTypeVar", bound="JSONDeSerializable")
```

Comment:
To be compliant with the other generics introduced in the project, could you rename this TypeVar into `GenericJSONDeSerializable` ?

Reply:
Certainly! done in last commit.